mlflow/store/ftp_artifact_repo.py

```diff
@@ -0,0 +1,124 @@
+import ftplib
+import os
+from ftplib import FTP
+
+from six.moves import urllib
+
+from mlflow.entities.file_info import FileInfo
+from mlflow.store.artifact_repo import ArtifactRepository
+from mlflow.utils.file_utils import TempDir, build_path, get_relative_path
+
+
+class FTPArtifactRepository(ArtifactRepository):
+    """Stores artifacts as files in a remote directory, via sftp."""
+
+    def __init__(self, artifact_uri, client=None):
+        self.uri = artifact_uri
+        parsed = urllib.parse.urlparse(artifact_uri)
+        self.config = {
+            'host': parsed.hostname,
+            'port': 21 if parsed.port is None else parsed.port,
+            'username': parsed.username,
+            'password': parsed.password
+        }
+        self.path = parsed.path
+
+        if self.config['host'] is None:
+            self.config['host'] = 'localhost'
+
+        if client:
```

Comment:
Since `self.ftp` holds state that's mutated by the implementation (e.g. by calling `self.ftp.cwd`), I believe the current implementation isn't thread-safe. 

Could we actually add a `get_ftp_client(self)` helper method that returns a new FTP client / is used to get an isolated FTP client in each method?

Reply:
I have add get_ftp_client(self) to return a new FTP client when be called.